kogitoq/kogito-common/src/main/java/life/genny/kogito/common/messages/SendAllMessages.java

```diff
@@ -0,0 +1,150 @@
+import org.jboss.logging.Logger;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SendAllMessages extends MessageSendingStrategy {
+
+    @Inject
+    SearchUtils searchUtils;
+
+    private String productCode;
+    private String milestoneCode;
+    private BaseEntity coreBE;
+
+    public SendAllMessages(String productCode, String milestoneCode, BaseEntity coreBE) {
+        this.productCode = productCode;
+        this.milestoneCode = milestoneCode;
+        this.coreBE = coreBE;
+    }
+
+    public SendAllMessages(String milestoneCode, String coreBeCode) {
+        this.milestoneCode = milestoneCode;
+        this.coreBE = beUtils.getBaseEntity(userToken.getProductCode(), coreBeCode);
+        this.productCode = this.coreBE.getRealm();
+    }
+
+    @Override
+    public void sendMessage() {
+        SearchEntity searchEntity = new SearchEntity("SBE_MILESTONE_MESSAGES",
+                "Fetch All Messages associated with milestone Code")
+                .addFilter("PRI_CODE", SearchEntity.StringFilter.LIKE, "MSG_%")
+                .addFilter("PRI_MILESTONE", SearchEntity.StringFilter.LIKE, "%\"" + milestoneCode.toUpperCase() + "\"%")
+                .setPageStart(0)
+                .setPageSize(100);
+
+        searchEntity.setRealm(productCode);
+
+        List<String> messageCodes = searchUtils.searchBaseEntityCodes(searchEntity);
+
+        if (messageCodes != null) {
+            log.info("messages : " + messageCodes.size());
+            for (String messageCode : messageCodes) {
+                log.info("messageCode : " + messageCode);
+                BaseEntity message = beUtils.getBaseEntityByCode(messageCode);
+                // Construct a contextMap
+                Map<String, String> ctxMap = new HashMap<>();
+                String recipientBECode = null;
+
+                // Determine the recipientBECode
+                String recipientLnkValue = message.getValueAsString("PRI_RECIPIENT_LNK");
+                if(recipientLnkValue != null) {
+                    recipientBECode = determineRecipientLnkValue(recipientLnkValue, ctxMap);
+                } else {
+                    log.error("NO PRI_RECIPIENT_LNK present");
+                    continue;
+                }
+
+                // Determine the sender
+                String senderLnkValue = message.getValueAsString("PRI_SENDER_LNK");
+                if (senderLnkValue != null) {
+                    ctxMap = determineSender(senderLnkValue, ctxMap);
+                } else {
+                    log.error("NO PRI_SENDER_LNK present");
+                    continue;
+                }
+
+                // Extract all the contexts from the core baseEntity LNKs
+                List<EntityAttribute> lnkEAs = coreBE.findPrefixEntityAttributes("LNK");
+                StringBuilder contextMapStr = new StringBuilder();
+                for (EntityAttribute ea : lnkEAs) {
```

Comment:
Use Java streams instead of traditional looping (serial stream).

Reply:
updated.